### PR TITLE
Add missing option validation

### PR DESCRIPTION
**Pull request by geeknoid**
null

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/extensions/pull/4260)

### CODE REVIEW

File: src/Libraries/Microsoft.Extensions.Telemetry.Abstractions/Latency/Registration/LatencyRegistryExtensions.cs [hunk 1]
```diff
@@ -79,7 +80,7 @@ private static void CheckNames(string[] names)
 
     private static void ConfigureOption(this IServiceCollection services, Action<LatencyContextRegistrationOptions> action)
     {
-        _ = services.AddOptions<LatencyContextRegistrationOptions>();
+        _ = services.AddValidatedOptions<LatencyContextRegistrationOptions, LatencyContextRegistrationOptionsValidator>();
```
Comment (xakep139) [inline src/Libraries/Microsoft.Extensions.Telemetry.Abstractions/Latency/Registration/LatencyRegistryExtensions.cs:83]: Do we need the options type to be public at all?

File: src/Libraries/Microsoft.Extensions.Telemetry.Abstractions/Microsoft.Extensions.Telemetry.Abstractions.csproj [hunk 2]
```diff
@@ -19,6 +19,7 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <ProjectReference Include="..\..\ToBeRemoved\Options.ValidateOnStart\Options.ValidateOnStart.csproj" />
```
Comment (joperezr) [inline src/Libraries/Microsoft.Extensions.Telemetry.Abstractions/Microsoft.Extensions.Telemetry.Abstractions.csproj:22]: I was under the impression that we were able to consume the .NET 8 version of this, is that not the case?

Comment (xakep139) [reply src/Libraries/Microsoft.Extensions.Telemetry.Abstractions/Microsoft.Extensions.Telemetry.Abstractions.csproj:22]: Good point

Comment (geeknoid) [reply src/Libraries/Microsoft.Extensions.Telemetry.Abstractions/Microsoft.Extensions.Telemetry.Abstractions.csproj:22]: Not yet, it's the last bit of ToBeRemoved functionality left. I think the PR went into dotnet/runtime yesterday or this morning, so this should be removed soon....

File: src/Libraries/Microsoft.Extensions.Telemetry.Abstractions/Latency/Registration/LatencyRegistryExtensions.cs (hunk 1 above)
Comment (geeknoid) [reply src/Libraries/Microsoft.Extensions.Telemetry.Abstractions/Latency/Registration/LatencyRegistryExtensions.cs:83]: I asked Ananth for feedback on that. Maybe we should hide it, we'll see what he ssys.

Comment (geeknoid) [reply src/Libraries/Microsoft.Extensions.Telemetry.Abstractions/Latency/Registration/LatencyRegistryExtensions.cs:83]: He's OOF, so it'll take a while. We can address this in a separate PR.

**Review by joperezr — APPROVED** [review]

**Pull request merged**